azurerm/internal/services/desktopvirtualization/virtual_desktop_workspace_application_group_association_resource.go

```diff
@@ -0,0 +1,182 @@
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"application_group_reference_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+		},
+	}
+}
+
+func resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.WorkspacesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Virtual Desktop Workspace <-> Application Group Association creation.")
+
+	workspaceID := d.Get("workspace_id").(string)
+	applicationGroupReferenceID := d.Get("application_group_reference_id").(string)
+
+	wsID, err := azure.ParseAzureResourceID(workspaceID)
+	if err != nil {
+		return err
+	}
+
+	agID, err := azure.ParseAzureResourceID(applicationGroupReferenceID)
+	if err != nil {
+		return err
+	}
+
+	workspaceName := wsID.Path["workspaces"]
+	resourceGroup := wsID.ResourceGroup
+	agName := agID.Path["applicationgroups"]
+
+	resourceID := fmt.Sprintf("%s|%s", workspaceID, applicationGroupReferenceID)
+
+	locks.ByName(workspaceName, "Microsoft.DesktopVirtualization/workspaces")
+	defer locks.UnlockByName(workspaceName, "Microsoft.DesktopVirtualization/workspaces")
+
+	locks.ByName(agName, "Microsoft.DesktopVirtualization/applicationgroups")
+	defer locks.UnlockByName(agName, "Microsoft.DesktopVirtualization/applicationgroups")
+
+	read, err := client.Get(ctx, resourceGroup, workspaceName)
+	if err != nil {
+		if utils.ResponseWasNotFound(read.Response) {
+			return fmt.Errorf("Retrieving Virtual Desktop Workspace %q (Resource Group %q) was not found", workspaceName, resourceGroup)
+		}
+
+		return fmt.Errorf("Retrieving Virtual Desktop Workspace for Association %q (Resource Group %q): %+v", workspaceName, resourceGroup, err)
+	}
+
+	refs := read.ApplicationGroupReferences
+
+	output := make([]string, 0)
+	output = append(output, *refs...)
+	if contains(output, applicationGroupReferenceID) {
+		return fmt.Errorf("Virtual Desktop Workspace Application Group Association already exists for Workspace %q and Application Group %q (Resource Group: %q)", workspaceName, agName, resourceGroup)
+	}
+	output = append(output, applicationGroupReferenceID)
+
+	read.ApplicationGroupReferences = &output
+
+	if _, err = client.CreateOrUpdate(ctx, resourceGroup, workspaceName, read); err != nil {
+		return fmt.Errorf("Updating Virtual Desktop Workspace Association for Application Group %q (Resource Group %q): %+v", workspaceName, resourceGroup, err)
+	}
+
+	d.SetId(resourceID)
+
+	return resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationRead(d, meta)
+}
+
+func resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	return nil
```

Comment:
How come this is nil and we don't actually make a read here?

we should be making a read to confirm the association still exists

Reply:
Yeah forgot to add this back. Will take a look at this